vars/job_monitoring.groovy

```diff
@@ -0,0 +1,75 @@
+import groovy.json.JsonSlurper
+
+/**
+ * Extracts the displayName from given build.
+ *
+ * @param response String Object containing the response from  previous HTTP GET request.
+ * @param upstream_status String Object indicates whether this is an upstream build.
+ */
+def build_info(response, upstream_status=null) {
+    def object = null
+    try {
+      def jsonSlurper = new JsonSlurper()
+      object = jsonSlurper.parseText(response)
+
+      if (upstream_status){
+        return object.description
+      }
+
+      def url = env.JENKINS_URL \
+             + object.actions.causes[0].upstreamUrl.join(", ") \
+             + object.actions.causes[0].upstreamBuild.join(", ") \
+             + "/api/json"
+
+      response = url.toURL().text
+      object = jsonSlurper.parseText(response)
+      } catch (Error e){}
```

Comment:
This here belongs to the try so it's not correctly indented. Delete two spaces pls.

Reply:
I will indent it properly, Which two spaces you mean,please?